src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs

```diff
@@ -232,40 +238,49 @@ public override async Task WriteAsync(byte[] buffer, int offset, int count, Canc
             }
         }
 
-        private void OnWrite()
+        private void InitializeCompression()
```

Comment:
```suggestion
        private void InitializeCompressionHeaders()
```
To clarify this doesn't set up the body.